build.gradle.kts

```diff
@@ -0,0 +1,56 @@
+plugins {
+    `java-library`
+    `maven-publish`
+}
+
+
+subprojects {
+    apply(plugin = "java-library")
+    apply(plugin = "maven-publish")
+
+    repositories {
+        mavenCentral()
+        mavenLocal()
+        gradlePluginPortal()
+    }
+
+
+//  This idiom completely mutes the "javadoc" task from java-library.
+//  Javadoc is still produced, but you won't get warnings OR build failures due to javadoc
+//  I decided to turn warning off because the amount of javadoc required for builder was too much.
+    tasks {
+        javadoc {
+            options {
+                (this as CoreJavadocOptions).addStringOption("Xdoclint:none", "-quiet")
+            }
+        }
+    }
+
+    java {
+        toolchain {
+            languageVersion.set(JavaLanguageVersion.of(17))
+        }
+        withSourcesJar()
+        withJavadocJar()
+    }
+
+    dependencies {
+        implementation("org.mitre:commons:0.0.54")
```

Comment:
These probably get leaked into your public API right? Commons and Guava at least should be `api()` not `implementation()` then.

Reply:
Correct, at least on commons